astropy/units/quantity.py

```diff
@@ -1317,12 +1317,15 @@ def _to_own_unit(self, value, check_precision=True):
             _value = value.unit.to(self.unit, value.value)
         except AttributeError:
             try:
-                _value = dimensionless_unscaled.to(self.unit, value)
-            except UnitsError as exc:
-                if _can_have_arbitrary_unit(value):
-                    _value = value
-                else:
-                    raise exc
+                _value = value.unit.to(self.unit, value.data)
```

Comment:
I don't think this is a good idea -- `Quantity` should not know too much about columns, and not assume a `data` property is OK. (E.g., `MaskedArray` also has a `.data` property, surely there are more classes that have it...).

Did you try the solution I gave in the issue? I do think that is the right way to go about this: one is allowed to use the `Quantity` constructor, and the constructor already knows how to deal with anything that has a `.unit` attribute.